addons/sale_mrp/models/account_move.py

```diff
@@ -12,7 +12,7 @@ def _stock_account_get_anglo_saxon_price_unit(self):
         so_line = self.sale_line_ids and self.sale_line_ids[-1] or False
         if so_line:
             bom = so_line.product_id.product_tmpl_id.bom_ids and so_line.product_id.product_tmpl_id.bom_ids[0]
-            if bom.type == 'phantom':
+            if bom.type == 'phantom' and (not bom.company_id or bom.company_id == so_line.company_id):
```

Comment:
```suggestion
            bom = so_line.product_id.product_tmpl_id.bom_ids.filtered(lambda b: not b.company_id or b.company_id == so_line.company_id)[:1]
            if bom and bom.type == 'phantom':
```
This way seems better.